src/setup-java.ts

```diff
@@ -13,6 +14,11 @@ async function run() {
 
     await installer.getJava(version, arch, jdkFile);
 
+    const username = core.getInput('username', {required: false});
+    const password = core.getInput('password', {required: false});
+
+    await auth.configAuthentication(username, password);
```

Comment:
I would add a `if` statement here early on to avoid even jumping into the method if one of the two inputs is not provided

Reply:
Ok, I moved it down lower.  

I wrapped the call in the control statement as requested.  I don't like this because its now an untested path. I have tests within the auth module for input validation because its simpler to test the smaller module than this file.